### ISSUE TITLE
Error when using only some astro-m2dx features in a file

### ISSUE THREAD

**Issue by astral4** [christian-hackyourshack/npm#7]
If you have the frontmatter feature enabled, you include a field in the frontmatter that would have been covered by a default, and you include an image with a relative import, an error is raised. Is this expected?

Example:

```ts
// config.ts

// The type of the frontmatter in the Markdown files.
export type Frontmatter = {
    title: string;
    description: string;
    layout: string;
};
```

```js
// tsconfig.json
{
  "extends": "astro/tsconfigs/strict",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@main": [
        "src/layouts/MainLayout.astro",
      ]
    },
  }
}
```

```yaml
# _frontmatter.yaml
layout: "@main"
```

```mdx
---
title: Inline Images
description: A demonstration of inlined images in Astro with MDX files.
---

## Example Gallery

![The answer to the ultimate question](./forty-two.jpg)
```

This example works, but when the `.mdx` file has a `layout` specified in the frontmatter (see below), an error is raised.

```mdx
---
title: Inline Images
description: A demonstration of inlined images in Astro with MDX files.
layout: ../layouts/MainLayout.astro
---

## Example Gallery

![The answer to the ultimate question](./forty-two.jpg)
```

**Comment by christian-hackyourshack** [comment]
Hi @astral4,

I tried to reproduce your error, but it seems to work for me. Do you have a reproduction repo for me, that I could try and could you paste the exact error message here? 

Thank you, Chris

P.S.: I tested with version 0.1.2, but I also just released a new version 0.2.0, maybe you want to try that with your setup. But beware: You explicitly have to activate all desired features, i.e. autoImports: true, frontmatter: true, ...

**Comment by astral4** [comment]
Thanks for the quick response! I tried making a minimal reproduction and re-launching the site that originally had the issue. I was not able to recreate the error, and even in the original site everything seems to be fine now. There are no problems in v0.2.0 either. I'm not sure what caused it, but it is probably not because of this package.
